lib/smf/smf.c

```diff
@@ -216,11 +362,101 @@ void smf_set_initial(struct smf_ctx *ctx, const struct smf_state *init_state)
 	if (init_state->entry) {
 		init_state->entry(ctx);
 	}
+#endif
 }
 
+#ifdef CONFIG_SMF_UML_HSM
+void smf_set_state(struct smf_ctx *const ctx, const struct smf_state *new_state)
+{
+	struct internal_ctx *const internal = (void *)&ctx->internal;
+
+	/*
+	 * It does not make sense to call set_state in an exit phase of a state
+	 * since we are already in a transition; we would always ignore the
+	 * intended state to transition into.
+	 */
+	if (internal->exit) {
+		LOG_WRN("Calling %s from exit action", __func__);
```

Comment:
`LOG_ERR`?

Reply:
It's LOG_WRN in classic, but I'm happy to make this an error.